igor-web/src/main/java/com/netflix/spinnaker/igor/helm/HelmMonitor.java

```diff
@@ -0,0 +1,207 @@
+  private final HelmProperties helmProperties;
+  private final Optional<EchoService> echoService;
+
+  @Autowired
+  public HelmMonitor(
+      IgorConfigurationProperties properties,
+      Registry registry,
+      DynamicConfigService dynamicConfigService,
+      DiscoveryStatusListener discoveryStatusListener,
+      Optional<LockService> lockService,
+      HelmCache cache,
+      HelmAccounts helmAccounts,
+      HelmProperties helmProperties,
+      Optional<EchoService> echoService,
+      TaskScheduler scheduler) {
+    super(
+        properties,
+        registry,
+        dynamicConfigService,
+        discoveryStatusListener,
+        lockService,
+        scheduler);
+    this.cache = cache;
+    this.helmAccounts = helmAccounts;
+    this.helmProperties = helmProperties;
+    this.echoService = echoService;
+  }
+
+  @Override
+  protected void initialize() {}
+
+  @Override
+  public void poll(boolean sendEvents) {
+    helmAccounts.updateAccounts();
+    helmAccounts.accounts.forEach(
+        account -> pollSingle(new PollContext(account.name, account.toMap(), !sendEvents)));
+  }
+
+  @Override
+  public PollContext getPollContext(String partition) {
+    Optional<HelmAccount> account =
+        helmAccounts.accounts.stream().filter(it -> it.name.equals(partition)).findFirst();
+    if (!account.isPresent()) {
+      throw new IllegalStateException(
+          String.format("Cannot find Helm account named '%s'", partition));
+    }
+    return new PollContext(account.get().name, account.get().toMap());
+  }
+
+  @Override
+  protected HelmPollingDelta generateDelta(PollContext ctx) {
+    final String account = ctx.partitionName;
+    log.info("Checking for new Helm Charts {}", kv("account", account));
+
+    List<HelmDelta> deltas = new ArrayList<>();
+
+    HelmIndex index = helmAccounts.getIndex(account);
+    if (index == null) {
+      log.error("Failed to fetch index {}", kv("account", account));
+    } else {
+      Set<String> cachedCharts = cache.getCharts(account);
+
+      // If we have no cache at all, do not fire any trigger events
+      // This is so we don't trigger every chart version if Redis dies
+      boolean eventable = !cachedCharts.isEmpty();
+
+      index.entries.forEach(
+          (key, charts) ->
+              charts.forEach(
+                  chartEntry -> {
+                    if (!StringUtils.isEmpty(chartEntry.digest)
+                        && !cachedCharts.contains(chartEntry.digest)) {
+                      deltas.add(
+                          new HelmDelta(
+                              chartEntry.name, chartEntry.version, chartEntry.digest, eventable));
+                    }
+                  }));
+    }
+
+    if (!deltas.isEmpty()) {
+      List<HelmDelta> eventableDeltas =
+          deltas.stream().filter(it -> it.eventable).collect(Collectors.toList());
+      log.info("Found {} new Helm charts {}", eventableDeltas.size(), kv("account", account));
```

Comment:
I think we should log the amount of deltas and eventableDeltas. Otherwise we will get logs like these on first cache, which could be misleading:
```
Checking for new Helm Charts myAccount
Found 0 new Helm charts myAccount
```